File: 268.missing-number.cpp
// @leet start
class Solution {
  public:
    int missingNumber(vector<int> &nums) {
        int answer = 0;
        int n = nums.size();
        for (int i = 0; i < n; i++) {
			answer ^= i ^ nums[i];
        }
        answer ^= n;
        return answer;
    }
};
// @leet end

File: 268.missing-number.js
// @leet start
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
	let n = nums.length;
	let answer = (n * (n + 1)) / 2 - sum(nums);
	return answer;
};
var sum = function(n) {
	let total = 0;
	for (let i = 0; i < n.length; i++) {
		total += n[i];
	}
	return total;
}
// @leet end
